### ISSUE TITLE
steps_per_epoch parameter in fit not working with tf.keras.utils.Sequence

### ISSUE THREAD

**Issue by dekromp** [tensorflow/tensorflow#28710]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): `Yes`
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): `Ubuntu 14.04`
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): `binary (conda)`
- TensorFlow version (use command below): `1.13.1`
- Python version: `3.6`
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
setting the `steps_per_epoch` parameter in `model.fit` (`tf.keras`) to for example 10 should only perform 10 updates per epoch. When passing numpy arrays as inputs to `model.fit` this works as expected, but when using a custom `tf.keras.utils.Sequence`  instance `steps_per_epoch` does not have any effect.

**Describe the expected behavior**
When passing `steps_per_epoch` to `model.fit` as a parameter when training on data fed via a `tf.keras.utils.Sequence` instance the model should be updated exactly `steps_per_epoch` times in every epoch.

**Code to reproduce the issue**
```python
import math

import numpy as np
import tensorflow as tf


class SomeFeeder(tf.keras.utils.Sequence):
    """A dummy Sequence.

    Slightly modified tf.keras.utils.Sequence example.
    """

    def __init__(self, x_set, y_set, batch_size):
            self.x, self.y = x_set, y_set
            self.batch_size = batch_size

    def __len__(self):
        return math.ceil(len(self.x) / self.batch_size)

    def __getitem__(self, idx):
        batch_x = self.x[idx * self.batch_size:(idx + 1) * self.batch_size]
        batch_y = self.y[idx * self.batch_size:(idx + 1) * self.batch_size]
        return batch_x, batch_y


# Simple dummy model.
input_x = tf.keras.Input((4,))
output = tf.keras.layers.Dense(1, activation='sigmoid')(input_x)
model = tf.keras.Model(inputs=[input_x], outputs=[output])
model.compile(
    optimizer=tf.keras.optimizers.SGD(lr=0.1), loss='binary_crossentropy')

# Dummy data.
x = np.random.rand(100, 4).astype(np.float32)
y = np.random.choice([0, 1], size=(100,)).astype(np.float32)
x = SomeFeeder(x, y, batch_size=4)

# Train the model with `steps_per_epoch=5`.
model.fit(x=x, epochs=10, steps_per_epoch=5)
```

**Other info / logs**
Produces:
```
Epoch 1/10
25/25 [==============================] - 0s 13ms/step - loss: 0.7041
Epoch 2/10
25/25 [==============================] - 0s 5ms/step - loss: 0.7007
Epoch 3/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6993
Epoch 4/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6981
Epoch 5/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6939
Epoch 6/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6954
Epoch 7/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6936
Epoch 8/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6914
Epoch 9/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6928
Epoch 10/10
25/25 [==============================] - 0s 5ms/step - loss: 0.6916
```
I would have expected `5/5` instead of `25/25`

**Comment by abhinavsagar** [comment]
Can reproduce this. Looks a bug to me.

**Comment by achandraa** [comment]
I was able to reproduce the mentioned output with TensorFlow version 1.13.1.

**Comment by qlzh727** [comment]
I can't reproduce this issue in tf.nightly, which indicate that this issue has been recently fixed, but hasn't reach release yet.

**Comment by qlzh727** [comment]
TF version = 1.14.1-dev20190523

Epoch 1/10
1/5 [=====>........................] - ETA: 0s - loss: 0.6458
5/5 [==============================] - 0s 39ms/step - loss: 0.7210
Epoch 2/10
1/5 [=====>........................] - ETA: 0s - loss: 0.9641
5/5 [==============================] - 0s 1ms/step - loss: 0.7426
Epoch 3/10
1/5 [=====>........................] - ETA: 0s - loss: 0.9364
5/5 [==============================] - 0s 1ms/step - loss: 0.7591
Epoch 4/10
1/5 [=====>........................] - ETA: 0s - loss: 0.6825
5/5 [==============================] - 0s 1ms/step - loss: 0.6975
Epoch 5/10
1/5 [=====>........................] - ETA: 0s - loss: 0.6900
5/5 [==============================] - 0s 1ms/step - loss: 0.6927
Epoch 6/10
1/5 [=====>........................] - ETA: 0s - loss: 0.6488
5/5 [==============================] - 0s 20ms/step - loss: 0.6489
Epoch 7/10
1/5 [=====>........................] - ETA: 0s - loss: 0.7134
5/5 [==============================] - 0s 1ms/step - loss: 0.7097
Epoch 8/10
1/5 [=====>........................] - ETA: 0s - loss: 0.7574
5/5 [==============================] - 0s 1ms/step - loss: 0.7197
Epoch 9/10
1/5 [=====>........................] - ETA: 0s - loss: 0.6375
5/5 [==============================] - 0s 1ms/step - loss: 0.6965
Epoch 10/10
1/5 [=====>........................] - ETA: 0s - loss: 0.6357
5/5 [==============================] - 0s 1ms/step - loss: 0.6624

**Comment by jashshah** [comment]
I am using TensorFlow version 1.13.1. Is there any work around for this?

**Comment by grouchysmurf** [comment]
Can confirm, ran a training run in 1.13.1 and had this issue, upgraded to 1.14.0 without doing anything else and now works correctly.

**Comment by zink-chimaera** [comment]
This bug is still present in 1.13.2.
